driver_4_lights.ino

```diff
@@ -1,225 +1,345 @@
+  digitalWrite(cameraShutter, LOW);
+  delay(shutterLag);
+  Serial.println("take my picture");
+  digitalWrite(cameraShutter, HIGH);
+  delay(shutterDelay);
+  digitalWrite(cameraShutter, LOW);
+  delay(shutterLag);
 }
 
 void led_set(uint8_t number, uint8_t setting) {
   if (setting == OFF) {
     digitalWrite(number, LOW);
-  } else
+  } else {
     digitalWrite(number, HIGH);
+  }
+}
+
+
+void doNoLights() {
+  led_set(ledTR, OFF);
+  led_set(ledTL, OFF);
+  led_set(ledBR, OFF);
+  led_set(ledBL, OFF);
+  Serial.println("Turning all lights off");
+}
+
+void doAllLights() {
+  led_set(ledTR, ON);
+  led_set(ledTL, ON);
+  led_set(ledBR, ON);
+  led_set(ledBL, ON);
+  delay(500);
+  if (buttonCamera == ON) {
+    takePicture();
+    Serial.println("picture taken for all lights");
+  }
+}
+
+void doTRLights() {
+  led_set(ledTR, ON);
+  led_set(ledTL, OFF);
+  led_set(ledBR, OFF);
+  led_set(ledBL, OFF);
+  delay (500);
+  if (buttonCamera == ON) {
+    takePicture();
+    Serial.println("picture taken for TR");
+  }
+}
+
+void doTLLights() {
+  led_set(ledTR, OFF);
+  led_set(ledTL, ON);
+  led_set(ledBR, OFF);
+  led_set(ledBL, OFF);
+  delay(500);
+  if (buttonCamera == ON) {
+    takePicture();
+    Serial.println("picture taken for TL");
+  }
+}
 
+void doBRLights() {
+  led_set(ledTR, OFF);
+  led_set(ledTL, OFF);
+  led_set(ledBR, ON);
+  led_set(ledBL, OFF);
+  delay(500);
+  if (buttonCamera == ON) {
+    takePicture();
+    Serial.println("picture taken for BR");
+  }
 }
-void camera_set(uint8_t number) {
-  digitalWrite(number, HIGH);
-  delay(3);
-  digitalWrite(number, LOW);
+
+void doBLLights() {
+  led_set(ledTR, OFF);
+  led_set(ledTL, OFF);
+  led_set(ledBR, OFF);
+  led_set(ledBL, ON);
+  delay(500);
+  if (buttonCamera == ON) {
+    takePicture();
+    Serial.println("picture taken for BL");
+  }
 }
 
 
+///////////////////////////////////////////////////... main loop
+void loop() {
+
+  //There is probably a function where you use state and millis() 
+  //    to decide when to turn indicateor light on, of or stay on
+  //    depending on state and timer.
+  lastIndicatorLightChange = handleIndicatorLED (state, lastIndicatorLightChange);
+
+  //Check state change
+  bool stateChange = debounceButtonState();
+  if (stateChange && buttonState == LOW) {
+    state = (state + 1) % 4; //cycle state 0, 1, 2, 3
+    lastIndicatorLightChange = handleIndicatorLED(state, lastIndicatorLightChange); //Indicate by IndicatorLED what state we are in
+    Serial.print("State = ");
+    Serial.println(state);
+    resetSystemState(); //todo set state to starting state again
+  }
+
+  //Handle the states
+  if (state == 1) { //stepping lights
+    bool cameraChange = debounceButtonCamera();
+    if (cameraChange && buttonCamera == LOW) {
+      //  ??? Needs mod 6, 0-5 'states' no, all, tr,tl,bt,bl
+      lightOnMode = (lightOnMode + 1) % 6; //0 no lights, 1 all, 2 TR on, 3 TL on, 4 BR on, 5 BL on
```

Comment:
You can create the last value in an enum to be SIZE and then do it like this:

lightOnMode = (lightOnMode + 1) % LightMode::SIZE;